src/assert.js

```diff
@@ -15,24 +15,34 @@ QUnit.assert = Assert.prototype = {
 		}
 	},
 
-	// Increment this Test's semaphore counter, then return a single-use function that
+	// Increment this Test's semaphore counter, then return a function that
 	// decrements that counter a maximum of once.
-	async: function() {
+	async: function( count ) {
 		var test = this.test,
-			popped = false;
+			popped = false,
+			acceptCallCount = count;
+
+		if ( typeof acceptCallCount === "undefined" ) {
+			acceptCallCount = 1;
+		}
 
 		test.semaphore += 1;
 		test.usedAsync = true;
 		pauseProcessing();
 
 		return function done() {
-			if ( !popped ) {
-				test.semaphore -= 1;
-				popped = true;
-				resumeProcessing();
+			acceptCallCount -= 1;
+			if ( acceptCallCount <= 0 ) {
+				if ( !popped ) {
+					test.semaphore -= 1;
+					popped = true;
+					resumeProcessing();
+				} else {
+					test.pushFailure( "Too many called the callback returned from `assert.async`",
```

Comment:
I would have used "Exceeded calls to the `assert.async` callback" instead. But that works too.